pkg/crc/machine/kubeconfig.go

```diff
@@ -0,0 +1,84 @@
+package machine
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/openshift/oc/pkg/helpers/tokencmd"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+)
+
+const (
+	adminContext     = "crc-admin"
+	developerContext = "crc-developer"
+	clusterName      = "api-crc-testing:6443"
+)
+
+func WriteKubeconfig(kubeadminPassword string) error {
+	kubeconfig := getGlobalKubeConfigPath()
+	dir := filepath.Dir(kubeconfig)
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		return err
+	}
+	// Make sure .kube/config exist if not then this will create
+	_, _ = os.OpenFile(kubeconfig, os.O_RDONLY|os.O_CREATE, 0600)
+
+	cfg, err := clientcmd.LoadFromFile(kubeconfig)
+	if err != nil {
+		return err
+	}
+
+	cfg.Clusters[clusterName] = &api.Cluster{
+		Server:                constants.DefaultAPIURL,
+		InsecureSkipTLSVerify: true,
+	}
+
+	if err := addContext(cfg, adminContext, "kubeadmin", kubeadminPassword); err != nil {
+		return err
+	}
+	if err := addContext(cfg, developerContext, "developer", "developer"); err != nil {
+		return err
+	}
+
+	if cfg.CurrentContext == "" {
+		cfg.CurrentContext = developerContext
+	}
+
+	return clientcmd.WriteToFile(*cfg, kubeconfig)
+}
+
+func addContext(cfg *api.Config, context, username, password string) error {
+	token, err := tokencmd.RequestToken(&restclient.Config{
+		Host: constants.DefaultAPIURL,
+		TLSClientConfig: restclient.TLSClientConfig{
+			Insecure: true,
+		},
+	}, nil, username, password)
+	if err != nil {
+		return err
+	}
+	cfg.AuthInfos[context] = &api.AuthInfo{
+		Token: token,
+	}
+	cfg.Contexts[context] = &api.Context{
+		Cluster:   clusterName,
+		AuthInfo:  context,
+		Namespace: "default",
+	}
+	return nil
+}
+
+// getGlobalKubeConfigPath returns the path to the first entry in KUBECONFIG environment variable
+// or if KUBECONFIG not set then $HOME/.kube/config
+func getGlobalKubeConfigPath() string {
+	pathList := filepath.SplitList(os.Getenv("KUBECONFIG"))
+	if len(pathList) > 0 {
+		// Tools should write to last entry in the KUBECONFIG file instead first one.
```

Comment:
`the last entry` ... `instead of the first one`

Reply:
done :+1: